shadows/framework/src/main/java/org/robolectric/shadows/ImageUtil.java

```diff
@@ -85,13 +95,32 @@ public static boolean writeToStream(
   private static String getFormatName(CompressFormat compressFormat) {
     switch (compressFormat) {
       case JPEG:
+        return FORMAT_NAME_JPEG;
       case WEBP:
       case WEBP_LOSSY:
       case WEBP_LOSSLESS:
-        return "jpg";
       case PNG:
-        return "png";
+        return FORMAT_NAME_PNG;
     }
     throw new UnsupportedOperationException("Cannot convert format: " + compressFormat);
   }
+
+  private static boolean needAlphaChannel(CompressFormat compressFormat) {
+    return !FORMAT_NAME_JPEG.equals(getFormatName(compressFormat));
+  }
+
+  private static int getBufferedImageType(Bitmap.Config config, boolean needAlphaChannel) {
+    switch (config) {
```

Comment:
nit: config can be null (e.g. gif I believe), so this needs a null check, because switching on null causing an NPE.